packages/field-base/src/input-slot-mixin.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { dedupingMixin } from '@polymer/polymer/lib/utils/mixin.js';
+import { SlotMixin } from './slot-mixin.js';
+
+const InputSlotMixinImplementation = (superclass) =>
+  class InputSlotMixinClass extends SlotMixin(superclass) {
+    static get properties() {
+      /**
+       * String used to define input type.
+       */
+      return {
+        type: {
+          type: String,
+          readOnly: true
+        }
+      };
+    }
+
+    get slots() {
+      return {
+        ...super.slots,
+        input: () => {
+          const native = document.createElement('input');
+          const value = this.getAttribute('value');
+          if (value) {
+            native.setAttribute('value', value);
+          }
+          const name = this.getAttribute('name');
+          if (name) {
+            native.setAttribute('name', name);
+          }
+          if (this.type) {
+            native.setAttribute('type', this.type);
+          }
+          return native;
+        }
+      };
+    }
+
+    /** @protected */
+    get _inputNode() {
```

Comment:
One thing I noted was that everything that uses `InputSlotMixin`, still has to wire up this input node with the `inputElement` property. Would it be possible to extend `InputSlotMixin` from `InputMixin` and then have the `InputSlotMixin` initialize the `inputElement` by itself? That would remove the need for callers of the mixin to "initialize" the `InputMixin`.

That would also allow to adapt `AriaLabelMixin` to use `inputElement`.

Reply:
The whole idea behind this PR is to split `InputMixin` and `InputSlotMixin` so we could use the following:

```html
<vaadin-combo-box-light>
  <vaadin-text-field slot="input"></vaadin-text-field>
</vaadin-cobo-box-light>
```

Unlike other components, the `-light` elements are not supposed to initialize the input element.
Instead they expect it to be provided to the slot manually. Maybe I should clarify this somewhere?

Reply:
Updated to make `InputSlotMixin` wrap the `InputMixin` and initialize the `inputElement` property.